### ISSUE TITLE
insights: Benchmark series load times

### ISSUE THREAD

**Issue by CristinaBirkel** [sourcegraph/sourcegraph#42403]
This issue is to be able to answer the question: how long does it take a series to load in the web app, and what are the biggest factors that influence this? At what point does it fall over? A couple of things to call out:

1. This will allow us to get a sense of what existing customers should expect, and also give us a baseline to test new loading models against.
2. This might also give us more insight into what areas could use the most improvement.

Some things that could influence the load times:
- How big the series is
- How many excluded repos there are for the user fetching the series
- How much data from other series is in the database

/cc @joelkw @felixfbecker @vovakulikov

**Comment by chwarwick** [comment]
Instead of trying to test this on k8s we could create benchmarks in go test code.  That would allow us to recreate a consistent environment so we could measure performance changes.

**Comment by coury-clark** [comment]
@leonore

**Comment by leonore** [comment]
I would timebox this to 2 days to benchmark in go code. If it doesn't work/we find it isn't representative we should pivot to k8s or the scaletesting environment and measure using that.

**Comment by chwarwick** [comment]
p0

**Comment by leonore** [comment]
closed by #44349. for testing this on the scaletesting env follow the following steps:
1. select a semi-random set of repos from the scaletesting instance (see the [handbook](https://handbook.sourcegraph.com/departments/engineering/dev/tools/scaletesting/) for accessing the db)
```sql
SELECT name from repo where name not like '%blank%' and name not like '%DELETED%' order by random() LIMIT 1000;
```
2. convert to a comma-separated list (use a website or other tool)
3. paste list into repo list
4. create insight (needs backfillerv2 to be enabled, toggle the option in site config if not created)
5. Add `&trace=1` to single insight page and follow trace to check loading time

[skeleton sheet](https://docs.google.com/spreadsheets/d/1Yp8loKBgD0oL_GSAj0uS7HD9xglpSOCrtpEISl2imIk/edit#gid=0)
